test/source-maps-test.ts

```diff
@@ -1,27 +1,48 @@
 import * as assert from "assert";
+import * as fs from "fs";
+import * as path from "path";
 
 import {transform} from "../src";
 
 describe("source maps", () => {
-  it("generates a simple line-based source map", () => {
-    const result = transform(
-      `\
+  const testCase = (simple: boolean): void => {
+    const source = `\
       import a from "./a";
       const x: number = 1;
       console.log(x + 1);
-    `,
-      {
-        transforms: ["imports", "typescript"],
-        sourceMapOptions: {compiledFilename: "test.js"},
-        filePath: "test.ts",
-      },
-    );
+    `;
+    const result = transform(source, {
+      transforms: ["imports", "typescript"],
+      sourceMapOptions: {compiledFilename: "test.js", simple},
+      filePath: "test.ts",
+    });
+    const simpleMappings = "AAAA;AACA;AACA;AACA";
     assert.deepEqual(result.sourceMap, {
       version: 3,
       sources: ["test.ts"],
       names: [],
-      mappings: "AAAA;AACA;AACA;AACA",
+      mappings: simple ? simpleMappings : result.sourceMap!.mappings,
       file: "test.js",
     });
+    if (!simple) {
+      if (process.env.WRITE_SOURCE_MAPS) {
+        const outDir = path.join(__dirname, "output");
+        fs.mkdirSync(outDir, {recursive: true});
+        result.sourceMap!.sourcesContent = [source];
+        let suffix = "//# sourceMapping";
+        suffix += `URL=test.js.map`;
+        fs.writeFileSync(path.join(outDir, "test.js"), `${result.code}\n${suffix}`);
+        fs.writeFileSync(path.join(outDir, "test.js.map"), JSON.stringify(result.sourceMap));
+      }
+      const {mappings} = result.sourceMap!;
+      assert.match(mappings, /^[^;]+(;[^;]+){3}$/); // 4 lines
```

Comment:
Could you actually just assert the entire string? It looks to me like it isn't so long, and I'd think of it like a snapshot test where any changes (e.g. changes to the column logic or a library upgrade) would be spot-checked, even though it's a bit opaque to someone reading the test.